drivers/SmartThings/matter-sensor/src/init.lua

```diff
@@ -93,28 +146,62 @@ local function check_for_battery(device)
     profile_name = profile_name .. "-pressure"
   end
 
+  if device:supports_capability(capabilities.rainSensor) then
+    profile_name = profile_name .. "-rain"
+  end
+
+  if device:supports_capability(capabilities.temperatureAlarm) then
+    profile_name = profile_name .. "-freeze"
+  end
+
+  if device:supports_capability(capabilities.waterSensor) then
+    profile_name = profile_name .. "-leak"
+  end
+
   if supports_battery_percentage_remaining(device) then
     profile_name = profile_name .. "-battery"
   end
 
+  if device:supports_capability(capabilities.hardwareFault) then
+    profile_name = profile_name .. "-fault"
+  end
+
+  local concatenated_preferences = supports_sensitivity_preferences(device)
+  profile_name = profile_name .. concatenated_preferences
+
   -- remove leading "-"
   profile_name = string.sub(profile_name, 2)
 
+  device.log.info_with({hub_logs=true}, string.format("Updating device profile to %s.", profile_name))
   device:try_update_metadata({profile = profile_name})
-  device:set_field(BATTERY_CHECKED, 1, {persist = true})
 end
 
 local function device_init(driver, device)
   log.info("device init")
-  if not device:get_field(BATTERY_CHECKED) then
-    check_for_battery(device)
-  end
   device:subscribe()
 end
 
 local function info_changed(driver, device, event, args)
   if device.profile.id ~= args.old_st_store.profile.id then
     device:subscribe()
+    set_boolean_device_type_per_endpoint(driver, device)
+  end
+  if not device.preferences then
+    return
+  end
+  for dt_name, info in pairs(BOOLEAN_DEVICE_TYPE_INFO) do
+    local dt_ep = device:get_field(dt_name)
+    if dt_ep and info.sensitivity_preference and (device.preferences[info.sensitivity_preference] ~= args.old_st_store.preferences[info.sensitivity_preference]) then
+      local sensitivity_preference = device.preferences[info.sensitivity_preference]
+      if sensitivity_preference == "2" then -- high
+        device:send(clusters.BooleanStateConfiguration.attributes.CurrentSensitivityLevel:write(device, dt_ep, device:get_field(info.sensitivity_max) - 1))
+      elseif sensitivity_preference == "1" then -- medium
+        local medium_sensitivity_level = math.floor((device:get_field(info.sensitivity_max) + 1) / 2)
+        device:send(clusters.BooleanStateConfiguration.attributes.CurrentSensitivityLevel:write(device, dt_ep, medium_sensitivity_level))
+      elseif sensitivity_preference == "0" then -- low
+        device:send(clusters.BooleanStateConfiguration.attributes.CurrentSensitivityLevel:write(device, dt_ep, MIN_SENSITIVITY_LEVEL))
```

Comment:
I'd move the definition of `MIN_SENSITIVITY_LEVEL` to right before it's used since it's only used in this one place

Reply:
Sounds good. However, there is a trend in the matter drivers to make every "magic number" variable a global, which is why it's like this.

Reply:
I think that makes sense when it's used in multiple places and some other select cases, but in this case it feels more confusing to define it so far from where it's used.

Reply:
Yeah, you're right. Now that I'm thinking about it, it used to be a global string that was used as a field, but then I realized that per spec the minimum will always be 0. I'll change it :+1: